src/Compilers/Core/MSBuildTask/MvidReader.cs

```diff
@@ -12,7 +13,10 @@ public static class MvidReader
 
         public static Guid ReadAssemblyMvidOrEmpty(Stream stream)
         {
-            return ReadAssemblyMvidOrEmpty(new BinaryReader(stream));
+            using (var reader = new BinaryReader(stream, new UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true), leaveOpen: true))
```

Comment:
>new UTF8Encoding(encoderShouldEmitUTF8Identifier: false, throwOnInvalidBytes: true) [](start = 57, length = 83)

`Encoding.UTF8` would be sufficient. The code doesn't decode any strings.

Reply:
Sure, thanks will change to `Encoding.UTF8` - I tried to retain existing semantics as I am not an expert in this part of the code base.